### DISCUSSION TITLE
Customize behavior on opening/closing tabs

### DISCUSSION THREAD

**Discussion by SlugCamargo** [Able-Bear-Software/tab-nodes-tree#10] [Ideas / Feature Request]
This may be a long shot, but: Is there any way you can add options to customize the browser's behavior on opening and/or closing tabs?

Right now I'm using two different plugins, one of them has any new tab (via link or ctrl+T) opening to the right of the one in focus; and another plugin that puts focus on the tab to the left of the one I'm closing. I've been having tabs work like that for what feels like forever in firefox (using sidebery at the moment), and to this day I can't make heads nor tails of the way chrome works by default.

The thing is, at least one of those extensions is dying any moment now because of chrome's new rules about plugins. And also, really, it would be fantastic to just have as many tab related custom options as possible in one plugin.

**Comment by zeroarst** [comment]
Hey @SlugCamargo, thanks for your feedback!

For opening tabs, when you use Ctrl+T to create a new tab, doesn't it already open to the right (or in Tab Nodes Tree, to the bottom as a child) and focus on it? I might have misunderstood your request.

As for closing tabs, are you referring to the [upcoming new Pro feature in version 1.2.0](https://github.com/Able-Bear-Software/tab-nodes-tree/discussions/8) that allows customization of which tab to activate after closing one?

**Reply by SlugCamargo** [reply]
> For opening tabs, when you use Ctrl+T to create a new tab, doesn't it already open to the right (or in Tab Nodes Tree, to the bottom as a child) and focus on it? 

OK, I disabled the extension I was using for that, and just realized that yours is doing the same thing. So that's taken care of, then.

On that subject, and this might not even be possible: Do you think this same behavior could be applied if a tab is opened _from outside of Chrome_? For example, I work with an excel book which has links to documents stored in a sharepoint - I click on a cell, and the document is opened in the browser; BUT! in this case the tab opens at the right end every time. The same happens if I click on a shortcut stored in my hard drive (a .URL file).


> As for closing tabs, are you referring to the https://github.com/Able-Bear-Software/tab-nodes-tree/discussions/8 that allows customization of which tab to activate after closing one?

Aye, I just received the notification for the upcoming update, and that sounds exactly like what I was aiming at. Looking forward to it!

**Reply by zeroarst** [reply]
> On that subject, and this might not even be possible: Do you think this same behavior could be applied if a tab is opened from outside of Chrome? For example, I work with an excel book which has links to documents stored in a sharepoint - I click on a cell, and the document is opened in the browser; BUT! in this case the tab opens at the right end every time. The same happens if I click on a shortcut stored in my hard drive (a .URL file).

This is definitely on my to-do list. I also find it frustrating when Chrome opens external links in new tabs at the very end. I'll be adding an option to address this in an upcoming version, possibly in 1.3.0.

Another feature I'm considering is auto-grouping tabs based on specific URLs. For example, you could have a `shopping` group and define URL patterns like `https://www.amazon.com/*`. Any new tabs matching that pattern would automatically be added to the `shopping` group. However, this feature is likely to be part of the Pro feature.

I'm currently working on fixing some issues in version 1.2.0 and aiming to have it released by tomorrow.

On another note, I've been tweaking some performance issues, and the improvements are significant, especially when you have hundreds of tabs open. Though maybe not everyone shares my habit of keeping so many tabs open, the difference is noticeable!

**Reply by SlugCamargo** [reply]
Ah, a dev after my own heart! 

Seriously, I can't express how much it warms my heart seeing a programmer in these days that cares for the same kind of crap I sometimes fear I'm the only person in the world who notices anymore. 


> This is definitely on my to-do list. I also find it frustrating when Chrome opens external links in new tabs at the very end. I'll be adding an option to address this in an upcoming version, possibly in 1.3.0.

I noticed this also happens when you open a tab from bookmarks, but I assume you're already aware.


> Another feature I'm considering is auto-grouping tabs based on specific URLs. For example, you could have a `shopping` group and define URL patterns like `https://www.amazon.com/*`. Any new tabs matching that pattern would automatically be added to the `shopping` group. However, this feature is likely to be part of the Pro feature.

This makes a lot of sense and it would actually save me one step in my usual workflow, since that's one of the things I use branches for.


> On another note, I've been tweaking some performance issues, and the improvements are significant, especially when you have hundreds of tabs open. Though maybe not everyone shares my habit of keeping so many tabs open, the difference is noticeable!

And indeed, one of the reasons why I need extensions like this is because I too am an appreciator of the ridiculously cluttered top bar. I'm still not in the hundreds, because I started using chrome relatively recently, but I am getting there xD

**Comment by zeroarst** [comment]
This issue should have been resolved in version 1.3.0 by incorporating the changes from #13.
